ui-events/src/pointer/mod.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2025 the UI Events Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Pointer Event Types
+
+mod buttons;
+
+pub use buttons::{PointerButton, PointerButtons};
+
+/// A unique identifier for the pointing device.
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct PointerId(i64);
+
+#[non_exhaustive]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+#[allow(missing_docs)]
+pub enum PointerEventType {
+    PointerOver,
+    PointerEnter,
+    PointerDown,
+    PointerMove,
+    PointerRawUpdate,
+    PointerUp,
+    PointerCancel,
+    PointerOut,
+    PointerLeave,
+    GotPointerCapture,
+    LostPointerCapture,
+}
```

Comment:
Everything else seems fine, but this one structure feels odd to me. I assume it shadows the web [`PointerEvent.type`](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent#pointer_event_types) property, but I'm not sure how we'd use it in Rust GUI contexts.

I'm assuming the goal of this library is to provide vocabulary types for crates like winit, masonry, iced, etc.

If that's the case, then as a masonry maintainer I don't know what I'd do with that enum. I wouldn't expect winit to give me `LostPointerCapture` events or even `PointerEnter`/`PointerLeave` events, because I don't expect winit to know anything about widget boundaries and pointer capture.

Reply:
I've removed this for now.

Reply:
But these events are for representing common scenarios that people would then re-use ... so your objection isn't quite valid.

But this being an enum vs a set of structs or a different type of enum ... that's worth discussing and handling differently.